diff --git a/c/asc.c b/c/asc.c
index 39da629..ee05a82 100644
--- a/c/asc.c
+++ b/c/asc.c
@@ -19,7 +19,6 @@
 	HA ASC method
 ***********************************************************************/
 
-#include <malloc.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include "ha.h"
diff --git a/c/cpy.c b/c/cpy.c
index 73f7b83..534d5f8 100644
--- a/c/cpy.c
+++ b/c/cpy.c
@@ -20,7 +20,6 @@
 ***********************************************************************/
 
 #include <stdio.h>
-#include <malloc.h>
 #include "ha.h"
 #include "cpy.h"
 #include "haio.h"
diff --git a/c/hsc.c b/c/hsc.c
index 47c06db..5380b56 100644
--- a/c/hsc.c
+++ b/c/hsc.c
@@ -19,7 +19,6 @@
 	HA HSC method
 ***********************************************************************/
 
-#include <malloc.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include "ha.h"
diff --git a/nix/machine.c b/nix/machine.c
index 06ae0db..08920a1 100644
--- a/nix/machine.c
+++ b/nix/machine.c
@@ -19,6 +19,7 @@
 	HA *nix specific routines
 ***********************************************************************/
 
+#include <string.h>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
@@ -333,7 +334,7 @@ void md_listdat(void) {
 
 char *md_timestring(unsigned long t) {
     
-    static char ts[22];
+    static char ts[34];
     struct tm *tim;	
     
     tim=localtime((long *)&t);
diff --git a/nix/machine.h b/nix/machine.h
index e2552a4..19fde4a 100644
--- a/nix/machine.h
+++ b/nix/machine.h
@@ -24,7 +24,7 @@
 #include <sys/stat.h>
 #include <signal.h>
 
-#define BETA "\337"
+#define BETA "ö"
 typedef short S16B;
 typedef unsigned short U16B;
 typedef int S32B;
